dsa-23au/java-dsa/sorting/src/main/java/dev/codewithfriends/NearestNeighbor.java

```diff
@@ -27,9 +27,20 @@ public static Point findNearestNeighborBruteForce(Point[] allPoints, Point targe
 
     public static Point findNearestNeighborHelper(Point[] allPoints, int startCoord, int endCoord, Point target) {
         return null;
+
+        if (endCoord == startCoord) {
+            return allPoints[startCoord];
+        } 
+        
+        else {
+            int centerCoord = findNearestNeighborHelper(startCoord, allPoints , endCoord, target); 
+            Point leftclosesPoint = findNearestNeighborHelper(startCoordCoord, allPoints , endCoord, target);
+            Point rightclosesPoint = findNearestNeighborHelper(startCoordCoord, allPoints , endCoord, target);    // right half
```

Comment:
These two calls to findNearestNeighborHelper() appear to use the same arguments. Wouldn't the leftclosesPoint and the rightclosesPoint be the same value as the method is being called on the same chunk of the array?

Reply:
I agree, the two recursive calls should divide up the range of coordinates of `startCoord` and `endCoord` of its parent.